### PR TITLE
Move compiler API to ballerina-lang

### PR DESCRIPTION
**Pull request by pubudu91**
## Purpose
$subject. 

## Approach
- Moved all the source code in the `main` directory directly to `ballerina-lang` module
- Renamed the `ballerina-compiler-api` module to `ballerina-compiler-api-test` and moved it to the tests directory

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: compiler/ballerina-compiler-api/src/main/java/module-info.java [hunk 1]
```diff
@@ -1,4 +0,0 @@
-module org.ballerina.compiler {
-    requires io.ballerina.tools.api;
```
Comment (nadeeshaan) [inline compiler/ballerina-compiler-api/src/main/java/module-info.java:2]: With this we can directly access the BallerinaSemanticModel implementation as well as the SemanticModel API.

**Review by SupunS — COMMENTED** [review]
Don't we need to add the `io.ballerina.compiler.api`, `io.ballerina.compiler.api.symbols`, `io.ballerina.compiler.api.types` to the module-info.java?

btw, should we move impl classes also under the `io.ballerina.compiler.api` package? Because once we change the package names of the existing `org.ballerinalang.compiler`/`org.wso2.ballerinalang.compiler` to `io.ballerina.compiler`, it will conflict with this. I was thinking of something like:
```
io.ballerina.compiler.api
  |_symbols
  |_types
  |_impl
```
Then all the compiler-api related classes are under one sub package.

File: compiler/ballerina-compiler-api/src/main/java/module-info.java (hunk 1 above)
Comment (nadeeshaan) [reply compiler/ballerina-compiler-api/src/main/java/module-info.java:2]: Also, shall we add `exports io.ballerina.compiler.api.symbols;`, `exports io.ballerina.compiler.api.symbols;` and `exports io.ballerina.compiler.impl;` to the ballerina-lang's module-info?

**Review by rasika — APPROVED** [review]

**Comment by rasika** [comment]
Until we encapsulate the initiation of `BallerinaSemanticModel` class through the ProjectAPI; we should expose it to external modules through the `module-info.java` file. For instance; `language-server-core` module requires to do`SemanticModel = new BallerinaSemanticModel()` in its module.

**Review by SupunS — APPROVED** [review]

**Pull request merged**